src/xgb-classifier/README.md

```diff
@@ -0,0 +1,59 @@
+# CVE Severity XGBoost Classifier
+
+XGBoost classifier for predicting Linux kernel CVE severity levels based on patch analysis and feature engineering.
+
+## Overview
+
+This classifier predicts CVE severity into three categories:
+- **IMPORTANT (0)**: High severity vulnerabilities requiring immediate attention
+- **MODERATE (1)**: Medium severity vulnerabilities  
+- **LOW (2)**: Low severity vulnerabilities
+
+## Data Pipeline & Feature Engineering
+
+The training data is constructed through a multi-step process:
+
+1. **CVE Data Collection**: Scrapes Linux kernel CVE metadata from security repositories
+2. **Commit & Patch Retrieval**: Fetches actual commit patches and details from the Linux kernel Git repository
+3. **Feature Engineering**: Analyzes patch content and commit messages to extract 48 binary features:
```

Comment:
minor nit: this is "feature extraction"  
"feature engineering" is the process of creating or curating features to use